tests/ceph_volume/test_ceph_volume_redeploy.py

```diff
@@ -0,0 +1,127 @@
+        if lvm_list.get(id, {})[0].get("type", {}) == "db":
+            osds[id] = lvm_list.get(id, {})[0].get("devices", {})
+    return osds
+
+
+def run(ceph_cluster, **kw):
+    """Re-deploy non-collocated OSDs with wrong dedicated DB size"""
+
+    # Get configs
+    get_configs()
+    config = kw.get("config")
+
+    # Get the installer and OSD nodes
+    installer = ceph_cluster.get_nodes(role="installer")[0]
+    osd_nodes = ceph_cluster.get_nodes(role="osd")
+
+    # Get device spec
+    device_spec = config.get("spec", {}).get("spec", {})[0].get("spec", {})
+
+    # Get cloud creadentials
+    cloud_type = config.get("cloud-type")
+    cloud_configs = get_cloud_credentials(cloud_type)
+    cloud = CloudProvider(cloud_type, **cloud_configs)
+
+    # Create volume for data device
+    size = device_spec.get("data_devices", {}).get("size", {}).replace("G", "")
+    _create_volume_in_bulk(nodes=osd_nodes, cloud=cloud, size=size, count=4)
+
+    # Create volume for db device
+    size = device_spec.get("db_devices", {}).get("size", {}).replace("G", "")
+    _create_volume_in_bulk(nodes=osd_nodes, cloud=cloud, size=size, count=2)
+
+    # Generate a yaml file
+    file = generate_bootstrap_config(node=installer, config=config.get("spec"))
+
+    # Create OSDs with yaml file
+    config = {"pos_args": [], "input": file}
+    CephAdm(nodes=installer).ceph.orch.apply(**config)
+
+    # Get the lvm list before OSD Zap
+    running_containers, _ = get_running_containers(
+        installer, format="json", expr="name=osd", sudo=True
+    )
+    container_ids = [item.get("Names")[0] for item in loads(running_containers)]
+    lvm_list = get_lvm_on_osd_container(container_ids[0], installer)
+
+    # Identify an OSD ID to perform
+    osd_ids = list(lvm_list.keys())
+
+    # Get OSD and device before zap
+    osd_db_before = _get_osd_db_id(osd_ids=osd_ids, lvm_list=lvm_list)
+
+    # Perform osd zap
+    conf = {"zap": True, "force": True}
+    for osd_id in osd_ids:
+        osd_rm = CephAdm(osd_nodes[0]).ceph.orch.osd.rm(osd_id=osd_id, **conf)
+        if not osd_rm:
+            raise RemoveCephVolumeError("Failed to remove osd")
+
+    # Wait until the rm operation is complete
+    timeout, interval = 300, 6
+    for w in WaitUntil(timeout=timeout, interval=interval):
+        conf = {"format": "json"}
+        out = CephAdm(osd_nodes[0]).ceph.orch.osd.rm(status=True, **conf)
+        if "No OSD remove/replace operations reported" in out:
+            break
+    if w.expired:
+        raise RemoveCephVolumeError("Failed to perform osd rm operation. Timed out!")
+
+    # Get the lvm list afer OSD zap
+    lvm_list = get_lvm_on_osd_container(container_ids[0], osd_nodes[0])
+
+    # Get OSD and device after zap
+    osd_db_after = _get_osd_db_id(osd_ids=osd_ids, lvm_list=lvm_list)
+
+    # Validate if db device has changed
+    if osd_db_before != osd_db_after:
+        raise RemoveCephVolumeError("Faild to re-deploy non-collocated OSD")
+    else:
```

Comment:
This else statement is not required here

Reply:
Okay, I'll remove it.